### PR TITLE
[ADD] contract_layout_category_hide_detail: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
This module allows you to check/uncheck 'Show details' and 'Show subtotal' flags for *Recurring invoices section lines* in Customer Contracts and Supplier Contracts so that Invoices are generated with this information.

Depends on:
- [x] https://github.com/OCA/contract/pull/486
- [x] https://github.com/OCA/sale-reporting/pull/87

Cc @Tecnativa TT22935

### CODE REVIEW

File: contract_layout_category_hide_detail/README.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to *Invoicing -> Customers -> Customer Contracts* and create
+   a new *Quotation*.
```
Comment (pedrobaeza) [inline contract_layout_category_hide_detail/README.rst]:
```suggestion
   a new *Contract*.
```

File: contract_layout_category_hide_detail/README.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+#. In *Recurring Invoices* tab add a section line.
+#. Add some product lines.
+#. Add another section line, but this time, click on the *plus-circle*
```
Comment (pedrobaeza) [inline contract_layout_category_hide_detail/README.rst]: another? there's no section for now.

File: contract_layout_category_hide_detail/tests/test_contract_layout_category_hide_detail.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from odoo.tests import common
+
+
+class TestContractLayoutCategoryHideDetail(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestContractLayoutCategoryHideDetail, cls).setUpClass()
+
+    # @classmethod
```
Comment (pedrobaeza) [inline contract_layout_category_hide_detail/tests/test_contract_layout_category_hide_detail.py]: You can omit tests for now and put it in the ROADMAP as we are over-passed on budget.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Aren't translations of the widgets required?

File: contract_layout_category_hide_detail/README.rst (hunk 2 above)
Comment (ernestotejeda) [reply contract_layout_category_hide_detail/README.rst]: Two spets before users have to create a secction: 
```In *Recurring Invoices* tab add a section line.```  https://github.com/OCA/contract/pull/489/files#diff-ae6e52733f87c5411e73b799af389834R45

**Comment by ernestotejeda** [comment]
changes done @pedrobaeza

**Review by chienandalu — CHANGES_REQUESTED** [review]
Rebase to get dependencies

**Comment by Tardo** [comment]
Check Travis

File: contract_layout_category_hide_detail/README.rst (hunk 2 above)
Comment (chienandalu) [reply contract_layout_category_hide_detail/README.rst]: I'd say that what Pedro means is that in your instructions you didn't add a section before (I'm following a functional test and it was confusing :sweat_smile: )

File: contract_layout_category_hide_detail/README.rst [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+   icon |plus-circle-icon| on your right. That icon will be replaced by
+   *minus-circle* icon |minus-circle-icon|. That mean *Show subtotal* field is
+   set to False.
+#. Add some product lines.
+#. Add another section line and click on the *eye* icon |eye-icon| on your
+   right. That icon will be replaced by *eye-slash* icon |eye-slash-icon|.
+   That mean *Show details* field is set to False.
```
Comment (chienandalu) [inline contract_layout_category_hide_detail/README.rst]: Maybe you should point that these settings can be set in the pop-up form when you click on the section line

**Review by chienandalu — COMMENTED** [review]
Somehow when the invoice was generated, the sections got messed up:
image

File: contract/models/contract.py [hunk 5]
```diff
@@ -416,7 +416,7 @@ def _get_lines_to_invoice(self, date_ref):
             lambda l: not l.is_canceled
             and l.recurring_next_date
             and l.recurring_next_date <= date_ref
-        )
+        ).sorted('sequence')
```
Comment (pedrobaeza) [inline contract/models/contract.py]: Why this?

Comment (ernestotejeda) [reply contract/models/contract.py]: I do this to guarantee that, when generating the invoice, each product line is under the corresponding section line and so what @chienandalu  commented in his review doesn't happen

Comment (pedrobaeza) [reply contract/models/contract.py]: But you should take the order of the model, not this one (`env['contract.line']._order`), and are you sure order is not preserved on filtering?

Comment (ernestotejeda) [reply contract/models/contract.py]: I was really wrong, the solution to the problem has nothing to do with reordering recordsets, haha. The 'sequence' field of the contract line was not being passed to create the invoice line. Thank you @pedrobaeza

File: contract/models/contract_line.py [hunk 6]
```diff
@@ -691,6 +691,7 @@ def _prepare_invoice_line(self, invoice_id=False, invoice_values=False):
         name = self._insert_markers(dates[0], dates[1])
         invoice_line_vals.update(
             {
+                'sequence': self.sequence,
```
Comment (pedrobaeza) [inline contract/models/contract_line.py]: Please remove this change from this PR, as it's included in the other.

**Review by pedrobaeza — APPROVED** [review]
Everything fixed and as @chienandalu's are attended, I'm merging

**Pull request merged**